### DISCUSSION TITLE
How to analyze gene marker from excel DEGs

### DISCUSSION THREAD

**Discussion by Deepesh123gupta** [satijalab/seurat#5039] [Ideas]
I have list of genes (i.e. 5000) in the excel with their log fold change and p_val_adjusted etc. What could be the wise approach to look for unique markers for each cluster?

I could manage to do that by looking at the heatmap and picking some random genes based on pct.1 and pct.2. Still, It takes time. Looking for some stepwise or smart suggestions.  

Thanks a lot.

**Comment by saketkc** [comment]
You could subset the `FindAllMarkers(obj, only.pos=T)` output to highlight the top 10 genes (weighed by log2FC or adjusted pvalue) for each cluster as shown in the [PBMC3k vignette](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html#finding-differentially-expressed-features-cluster-biomarkers-):

```
pbmc.markers %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(pbmc, features = top10$gene) + NoLegend()

```
